golang/cosmos/x/lien/types/key.go

```diff
@@ -0,0 +1,29 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	ModuleName = "lien"
+
+	StoreKey = ModuleName
+)
+
+var (
+	LienByAddressPrefix = []byte{0x01}
+)
```

Comment:
I don't think you need a prefix in this case.  The only reason the "storage" uses it is because the root node's key is an just an empty `[]byte{}`, which cannot be represented in the KVStore.  So, I hacked around their hack by adding a prefix to all the keys.

Reply:
Huh, I thought it was to preserve the ability to add different data in the same store in the future. In fact, look at all the different prefixes used in `cosmos-sdk/x/staking/types/keys.go`. I'd rather keep the prefix if it's not hurting anything.

Reply:
A key prefix is a hack that we should avoid if possible.  If you want to add different data in the same store, there's a better pattern:

```
import (
        "github.com/cosmos/cosmos-sdk/store/prefix"
)

var (
         subPrefix = []byte("some-kinda-prefix")
)

// ... and in a keeper function:
        store := ctx.KVStore(k.storeKey)
        subStore := prefix.NewStore(store, subPrefix)
```

Reply:
Groovy. Leaving the `address <--> key` conversion API, but no prefix.